src/library/trackprocessing.cpp

```diff
@@ -0,0 +1,108 @@
+#include "library/trackprocessing.h"
+
+#include <QProgressDialog>
+#include <QThread>
+
+#include "library/trackcollection.h"
+#include "library/trackcollectionmanager.h"
+#include "util/logger.h"
+
+namespace mixxx {
+
+namespace {
+
+const Logger kLogger("ModalTrackBatchProcessor");
+
+} // anonymous namespace
+
+int ModalTrackBatchProcessor::processTracks(
+        const QString& progressLabelText,
+        TrackCollectionManager* pTrackCollectionManager,
+        TrackPointerIterator* pTrackPointerIterator) {
+    DEBUG_ASSERT(pTrackCollectionManager);
+    DEBUG_ASSERT(pTrackPointerIterator);
+    DEBUG_ASSERT(QThread::currentThread() ==
+            pTrackCollectionManager->thread());
```

Comment:
VERIFY_OR_DEBUG_ASSERT and return early if any of these assertions are violated?

Reply:
No, these are mandatory asserts. Especially the threading requirements must never ever be violated.